cloud/services/groups/service.go

```diff
@@ -30,7 +31,11 @@ type Service struct {
 // GroupScope defines the scope interface for a disk service.
 type GroupScope interface {
 	logr.Logger
-	azure.ClusterDescriber
```

Comment:
This is too broad of an interface to expect here.

Reply:
I agree. I think this applies to some of the other services as well though. Maybe we should have a more focused interface for these things that every single service needs to know about and then have a broader one that includes the Network stuff that gets passed into the machine scope.

Reply:
See https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/824

@devigned do you mind reverting this change for now so we can deal with it in all services at once with #824?

Reply:
That means I'll have to add a bunch of superfluous funcs to the managed control plane scope. Which is better?

Reply:
Ok in that case let's just keep the change, will work on top of it

Reply:
done and done

Reply:
fwiw I think constraining the scope at site of use is more correct, network describer should only go into the top level machine/cluster scope (or lower level scopes that require all components)